### PR TITLE
cilium: add note into upgrade guide and perf guide about netkit enablement

### LINKED ISSUE #33291: docs: upgrade guide / perf guide note on netkit wrt inplace upgrade

### PR DESCRIPTION
**Pull request by borkmann**
See commits.

Closes: https://github.com/cilium/cilium/issues/33291

### CODE REVIEW

File: Documentation/operations/performance/tuning.rst [hunk 1]
```diff
@@ -91,6 +101,16 @@ in any of the Cilium Pods and look for the line reporting the status for
 "Device Mode" which should state "netkit". Also, ensure to have eBPF host
 routing enabled - the reporting status under "Host Routing" must state "BPF".
 
+.. note::
+    In-place upgrades by just enabling netkit on an existing cluster is not
```
Comment (ghost) [inline Documentation/operations/performance/tuning.rst]:
```suggestion
    In-place upgrade by just enabling netkit on an existing cluster is not
```

File: Documentation/operations/performance/tuning.rst [hunk 2]
```diff
@@ -165,6 +185,16 @@ than 64k it will try to decrease them.
 
 BIG TCP doesn't require network interface MTU changes.
 
+.. note::
+    In-place upgrades by just enabling BIG TCP on an existing cluster is currently
```
Comment (ghost) [inline Documentation/operations/performance/tuning.rst]:
```suggestion
    In-place upgrade by just enabling BIG TCP on an existing cluster is currently
```

File: Documentation/operations/performance/tuning.rst [hunk 3]
```diff
@@ -222,6 +252,16 @@ than 64k it will try to decrease them.
...
```
Comment (ghost) [inline Documentation/operations/performance/tuning.rst]:
```suggestion
    In-place upgrade by just enabling BIG TCP on an existing cluster is currently
```

File: Documentation/operations/performance/tuning.rst [hunk 4]
```diff
@@ -412,6 +452,18 @@ BBR also needs eBPF Host-Routing in order to retain the network packet's socket
 association all the way until the packet hits the FQ queueing discipline on the
 physical device in the host namespace.
 
+.. note::
+    In-place upgrades by just enabling BBR on an existing cluster is not possible
```
Comment (ghost) [inline Documentation/operations/performance/tuning.rst]:
```suggestion
    In-place upgrade by just enabling BBR on an existing cluster is not possible
```

File: Documentation/operations/performance/tuning.rst [hunk 5]
```diff
@@ -18,6 +18,16 @@ Recommendation
 The default out of the box deployment of Cilium is focused on maximum compatibility
 rather than most optimal performance. If you are a performance-conscious user, here
 are the recommended settings for operating Cilium to get the best out of your setup.
+
+.. note::
+    In-place upgrades by just enabling the config settings on an existing
```
Comment (ghost) [inline Documentation/operations/performance/tuning.rst]:
```suggestion
    In-place upgrade by just enabling the config settings on an existing
```

File: Documentation/operations/performance/tuning.rst [hunk 6]
```diff
@@ -18,6 +18,16 @@ Recommendation
...
+    cluster is not possible since these tunings chance the underlying datapath
```
Comment (ghost) [inline Documentation/operations/performance/tuning.rst]:
```suggestion
    cluster is not possible since these tunings change the underlying datapath
```

**Review by ghost — APPROVED** [review]
Left some grammar nitpicking comments, but LGTM.

**Review by lmb — APPROVED** [review]

**Pull request merged**